department-of-veterans-affairs/caseflow
# Intake Form Step 2.2: Review Tasks & Appeals - Tasks not related to an Appeal

Resolves [Intake Form Step 2.2 -- APPEALS-29416](https://jira.devops.va.gov/browse/APPEALS-29416)

# Description
Work performed on the second step of the correspondence intake queue. A new section "Tasks not related to an Appeal" was added to the page. The User may now select "+ Add Tasks" button to open a "New Tasks" section. Added logic to disable the continue button when information on the new task has not been filled out yet. This button will be reenabled if the task is removed or filled out completely. User is prevented from entering only a space character in the text field. Spec tests were written to test the logic of the new functionality on this page. 

## Acceptance Criteria
- [ ] The mail team user is able to access a section titled "Tasks not Related to an Appeal"
- Paragraph text appears below the title

- [ ] The mail team user is able to click an "+add tasks" button
- The user can add additional tasks to correspondence by selecting the "+add tasks" button again
- Two tasks is the limit for the user
- The "+add Tasks" button is grayed out when two tasks have been selected

- [ ] When the mail team user clicks the "+add tasks" button" a new section appears titled "New Tasks"

- [ ] The mail team user is able to click a dropdown with the below tasks
- CAVC Correspondence
- Congressional interest
- Death certificate
- FOIA request
- Other motion
- Power of attorney-related
- Privacy act request
- Privacy complaint
- Status inquiry

- [ ] The mail team user is required to add notes
- This is a text box
- There is paragraph text above the text box

- [ ] The mail team user is able to remove tasks when clicking the "remove tasks" link

- [ ] The continue button is inactive if there is an incomplete task that has been selected
- The dropdown has not been selected or the text box is not filled out

## Testing Plan

1. Navigate to /queue/correspondence/ and choose a veteran. Then select "Create record". Similarly, you may also navigate to /queue/correspondence/0c77d6d2-c19f-4dbb-8e79-919a4090ed33/intake
2. Verify "No" is selected on the "Add Related Correspondence" page (step 1), then select Continue. 
3. Verify the "Review Tasks & Appeals" page shows (step 2).
4. Select the "+ Add tasks" button, and verify new section "New Tasks" appears with one task box to fill in. Verify the continue button is disabled at this step.
5. Fill in the options for the new task. Make note of the choice from the dropdown. Verify continue button is still disabled if only one option is filled in.
6. Remove text from the text box and verify that the continue button is disabled again. Click the back button, and verify that the continue button is enabled on the first step to move back to the second step.
7. Click the continue button to proceed back to the second step, and verify that the dropdown choice from before is still present.
8. Click the "+ Add tasks" button again, verifying the continue button disables, and then fill out information for the second task. Make note of the information on the second task, and click the "Remove task" button on the first task. Verify the information from the second task is now the only information present. 
9. Continue to play around with the UI moving forward and backward during the process to make sure the information in each task is still present when returning to the step.


# Frontend
## User Facing Changes
Before:
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/a581c03d-1bd8-4080-af20-c4efc5930b5b)
---
After:
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/585e0810-c4a8-49ee-8183-0487be817e35)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/46821eea-8d42-444b-9de5-33f23f136494)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/75d3f46b-8b04-497e-837e-5ce307134ada)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/c8f15a36-22dd-4988-854e-b31288e08de5)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/74e70830-d843-4e48-b7fb-55bc0a1c22fc)
